### ISSUE TITLE
_variable_description meta tag not updating when changed in database

### ISSUE THREAD

**Issue by rcojoe** [woocommerce/woocommerce#18773]
## Prerequisites

- [ x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Create a variable product.  In one of the variants,  set the variable_description field (Description label) to "Widget". Go to bottom of variant area and save changes.
2. In the database,  find the wp_postmeta table and search for the new entry for the above variant item named "Widget".  The tag _variation_description for the post ID value should be "Widget".
3. In the database, modify the _variation_description to "Widget 123".
4. CLear all caches. Then go back into the admin product page and open up the new product you created in step 1.  Go to the variant that you typed in "Widget" into the variable_description field (Description label).  Observe that the variable_description still says "Widget", and not the new value "Widget 123".
5.  If you type in "Widget new" into the  variable_description field (Description label) and save changes at the bottom of the variant list, the new value will replace the modified "Widget 123" that you changed in the database.

## Expected/actual behavior

I expect the variable_description field to carry over to the  variable_description field (Description label) in the product page. 
It appears the  variable_description field (Description label) is reading from a different field, and writing to the variant_description in the database.  

## Isolating the problem

- [x ] This bug happens with only WooCommerce plugin active
- [ x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin here.
```
</details>

**Comment by kloon** [comment]
Hi, I have just tested this and cannot recreate. Once I change the db entry and flush the cache via wp-cli and reload the product edit page the modification shows.

How are you clearing your cache? It is possible that you are not clearing your object cache correctly hence why the old value will still show. If you have wp-cli installed try `wp cache flush`

Closing as this not reproducible from my end.

**Comment by rcojoe** [comment]
@kloon , You hit the nail on the head.  Wordpress caching was definitely part of my google escapade of dead ends to try to figure out what was going on.  I figured it was a cache, but I left out the search tearm "Object" which seems to make the difference.  I had cleared my content cache but didn't know about the object class cache.  This explain multiple issues I have had with loading product through the API and command line for the past year.  I always had to go back and do a product update to make all my variations show up.  Thank you for your insight and time spent on explaining and teaching me to fish. 👍
